### PR TITLE
More CMS tests for ROAs and Manifests

### PR DESCRIPTION
**Pull request by wibisono**
As suggested by ROS, add more test on CMS parts related to ROA/MFT.  Added property based tests for building (ROA/MFT) CMS, with generated KeyPairs, encode it, parse it back, and check if properties are still consistent.

### CODE REVIEW

File: src/test/java/net/ripe/rpki/commons/crypto/cms/roa/RoaCMSBuilderPropertyTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import static net.ripe.rpki.commons.crypto.cms.roa.RoaCmsParserTest.TEST_IPV4_PREFIX_2;
+import static net.ripe.rpki.commons.crypto.x509cert.X509CertificateBuilderHelper.DEFAULT_SIGNATURE_PROVIDER;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnitQuickcheck.class)
+public class RoaCMSBuilderPropertyTest {
+
+    @Property public void buildEncodedParseCheck(
+            KeyPair keyPair,
+            @InRange(min="1", max="65536")  long asNum
```
Comment (erikrozendaal) [inline src/test/java/net/ripe/rpki/commons/crypto/cms/roa/RoaCMSBuilderPropertyTest.java:52]: Shouldn't this also include 4-byte ASNs?

**Review by erikrozendaal — COMMENTED** [review]
The property based tests are very slow (complete `mvn test` goes from 6 seconds to 23 seconds on my laptop), most likely due to key pair generation. I don't think randomized key pairs add much (keys are generated on every run anyway), so I'm not sure if the tests are worth the extra CPU time etc. I'd focus on randomizing other parameters.

File: src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsBuilder.java [hunk 2]
```diff
@@ -103,10 +103,6 @@ public void addFile(String fileName, byte[] contents) {
         files.put(fileName, digestValue);
     }
 
-    public void addFileHash(String fileName, byte[] hash) {
```
[the remark this answers is not in the document]
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsBuilder.java:106]: @erikrozendaal We have checked that it's not used anywhere. What is the use case for it you have in mind?

[the remark this answers is not in the document]
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsBuilder.java:106]: Nevermind, the use case is very easy to imagine.

**Review by erikrozendaal — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsBuilder.java [hunk 3]
```diff
@@ -104,6 +104,7 @@ public void addFile(String fileName, byte[] contents) {
     }
 
     public void addFileHash(String fileName, byte[] hash) {
+        assert hash!= null && hash.length == 32: "Hash should be 32 bytes";
```
Comment (lolepezy) [inline src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCmsBuilder.java:107]: MUST ;)

**Review by lolepezy — APPROVED** [review]

**Pull request merged**